src/main/java/io/vertx/gsoc2023/qotd/QuoteOfTheDayVerticle.java

```diff
@@ -22,9 +29,32 @@ public void start(Promise<Void> startFuture) throws Exception {
       pgPool = setupPool(config);
       var router = setupRouter();
       var httpServer = createHttpServer(router);
+      
+      PgPool pool = setupPool(config);
+      router.get("/quotes").handler(ctx -> getHandler(pool, ctx));
+      
       return httpServer.listen(config.getInteger("httpPort", 8080)).<Void>mapEmpty();
     }).onComplete(startFuture);
   }
+  
+  private void getHandler(PgPool pool, RoutingContext ctx) {
+		HttpServerResponse response = ctx.response();
+		List<Quotes> receivedQuotes = new ArrayList<>();
+		pool.getConnection().compose(conn -> {
+			System.out.println("Got a connection from the pool");
+
+			return conn.query("SELECT * from quotes").execute().onComplete(rows -> {
+				for (Row row : rows.result()) {
+					Quotes quote = new Quotes();
+					quote.setId(row.getInteger(0));
+					quote.setQuote(row.getString(1));
+					quote.setAuthor(row.getString(2));
+					receivedQuotes.add(quote);
+				}
+			}).eventually(x -> conn.close());
+		}).onSuccess(x -> response.end(Json.encodePrettily(receivedQuotes)))
+				.onFailure(v -> response.end(v.getCause().getLocalizedMessage()));
```

Comment:
It is possible to simplify different things:

- execute query directly on the pool
- convert row directly to JSON (unnecessary POJO mapping)
- use future composition instead of future callbacks

This last point is important. There is no guarantee about the order of execution of callbacks, so you shouldn't use them to create a `List` created before executing the query.